src/platform/plugins/shared/data_view_editor/public/components/data_view_flyout_content_container.tsx

```diff
@@ -63,6 +65,17 @@ const DataViewFlyoutContentContainer = ({
         }
         saveResponse = editData;
       } else {
+        if (editData && isDuplicating) {
+          const editDataSpec = editData.toSpec();
+          dataViewSpec = {
+            ...editDataSpec,
+            id: undefined,
```

Comment:
```suggestion
            id: dataViewSpec.id,
```
It looks like due to this line, we can't set a custom ID when duplicating a data view. It allows me to set the input value, but it's cleared out on save:
<img width="400" src="https://github.com/user-attachments/assets/e7d9216f-ab11-4d73-af24-f50f2a930a53" />

I believe we already take care of clearing out the old ID when duplicating it here in Unified Search, so it might be ok to just use `dataViewSpec.id` for this (not sure about this though, should be tested): https://github.com/elastic/kibana/blob/1b769db71e7108733328b183b0cfbd7f719c41f5/src/platform/plugins/shared/unified_search/public/dataview_picker/change_dataview.tsx#L148